modules/ossm-config-external-ca.adoc

```diff
@@ -0,0 +1,75 @@
+// Module included in the following assemblies:
+//
+// * service_mesh/v2x/ossm-security.adoc
+
+:_content-type: CONCEPT
+[id="ossm-config-external-ca_{context}"]
+= Configuring {SMProductShortName} for an external certificate authority
+
+By default, {SMProductName} generates a self-signed root certificate and key and uses them to sign the workload certificates.  You can also use your own certificate provider to generate certificates.
+
+.Prerequisites
+
+* You must have the {SMProductName} 2.3 Operator installed.
+* You must have a version 2.3 `ServiceMeshControlPlane` deployed.
+
+The `ServiceMeshControlPlane` resource lets you configure an external certificate authority using the `security.certificateAuthority.custom.address` parameter. This enables {SMProductShortName} to integrate with external certificate authorities (CAs).
+
+[NOTE]
+====
+Red Hat does not support third-party certificate authorities.  Contact your CA vendor for support for your certificate provider.
+====
+
+.Example configuration for external certificate authority
+[source,yaml]
+----
+apiVersion: maistra.io/v2
+kind: ServiceMeshControlPlane
+metadata:
+  name: basic
+  namespace: istio-system
+spec:
+  security:
+    certificateAuthority:
+      cert-manager:
+        address: cert-manager-istio-csr.cert-manager.svc:443
+        pilotSecretName: istiod-tls
+        rootCAConfigMapName: "istio-ca-root-cert"
+      type: cert-manager
+----
+
+An additional requirement for this method is that the pilot server must have its tls context included in a secret that is generated by the external certificate authority.  The configuration for this secret is specified in the `security.certificateAuthority.pilotSecretName` parameter in the SMCP.
+
+.Example configuration for certificate
+[source,yaml]
+----
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: istio-ca
+spec:
+  isCA: true
+  duration: 2160h # 90d
+  secretName: istio-ca
+  commonName: istio-ca
+  subject:
+    organizations:
+      - cluster.local
+      - cert-manager
+  issuerRef:
+    name: selfsigned
+    kind: Issuer
+    group: cert-manager.io
+----
+
+//Not sure what to title this example?  Or even if we need to include it?
```

Comment:
Yes, this generates an `Issuer` named `istio-ca`  that uses the key/cert generated above (the `Certificate`).  I also think we need to include the `selfsigned` `Issuer` too, as it's used to create/sign the certificate above.

Of all of this, I think the flow is:
* Root CA - in this case a self-signed certificate generated using `Issuer.spec.selfSigned` (I think we need to add this for the example.)
* Istio CA - This is still the `istio-ca` `Secret` located in the `istio-system` namespace, but it is generated from the `istio-ca` `Certificate` (above).  This certificate will be used to sign certificates for the `istio-ca` `Issuer` configured below.

I *think* this is correct.

Reply:
@rcernich I've revised based on my current understanding of the workflow.  But the examples in 
https://github.com/rcernich/istio-operator/blob/f92e8142120a11c2b8ca0487907193fad2d11272/deploy/examples/cert-manager/selfsigned-ca.yaml

And 
https://github.com/rcernich/istio-operator/blob/f92e8142120a11c2b8ca0487907193fad2d11272/deploy/examples/cert-manager/smcp.yaml

Don't have matching values for the secret names.  Should they match?  (See my comment in the example).